docs/understanding-radicle/how-it-works.md

```diff
@@ -1,80 +1,488 @@
+    /// character
+    name: String,
+}
+
+struct Project {
+    /// A short name
+    name: String,
+
+    /// A slightly longer description (should fit in a headline)
+    description: Option<String>,
+
+    /// The default branch. "master" is assumed for git repositories if
+    /// unspecified.
+    default_branch: Option<String>,
+}
+
+```
+
+There are currently no restrictions on the length (in bytes) of the fields.
+Applications may add additional payload data, but must do so in a way which
+unambiguously preserves the shape of the above definitions. See
+[Serialisation][se].
+
+### Radicle URNs
+
+Identities are addressable within the Radicle Network by their stable
+identifier, encoded as a URN. Radicle URNs are syntactically and functionally
+equivalent to URNs as per [RFC8141][r8].
+
+The syntax of a Radicle URN is defined as follows:
+
+```rust
+"rad" ":" protocol ":" root [ "/" path ]
+
+```
+
+where:
+
+```rust
+protocol = "git"
+root     = MULTIBASE(MULTIHASH(id))
+path     = pct-encoded
+id       = BYTES
+
+```
+
+The `id` is the `root` field of a verified Identity, as specified previously.
+The `MULTIBASE` and `MULTIHASH` encodings are specified in [multibase][mu] and
+[multihash][mh], respectively. The preferred alphabet for the multibase encoding
+is [z-base32][zb]. pct-encoded is defined in [RFC3986][r3], and the equivalence
+rules as per [RFC8141][r8] apply.
+
+An example of a Radicle URN:
+`rad:git:hwd1yredksthny1hht3bkhtkxakuzfnjxd8dyk364prfkjxe4xpxsww3try`.
+
+In the Upstream client, Radicle URNs are called [**Radicle IDs**][ri].
+
+Read more [here][ur]
+
+
+### Delegations
+
+As described in [Data Model][dm] Radicle Link distinguishes two types of
+identities: personal and project. The first describes an person (usually denoted
+as a `peer`) in the system, while the second describes a software project
+(repository) on which one or more persons collaborate. Apart from their
+[payload][dp] types `T`, they differ in their delegations type `D`:
+
+Personal identities can only delegate to anonymous keys, while project
+identities may attach a personal identity to a key delegation. More formally:
+
+```rust
+type User<T> = Identity<T, HashSet<PublicKey>>;
+
+enum ProjectDelegation<U> {
+    Key(PublicKey),
+    User(User<U>),
+}
+
+type Project<T, U> = Identity<T, ProjectDelegation<U>>;
+
+```
+
+The `delegations` of a `Project` are also referred to as the project's
+**maintainers**.
+
+These delegations are used to verify the authenticity of data across the
+network. See more notes on Verification [here][vr].
+
+### Key Architecture
+
+With delegations, all identity in Radicle can be verifiable by public-key
+cryptography. Keys on Radicle are architected as such:
+
+* A `DeviceKey` is an Ed25519 keypair tied to a peer's device (`d`)
+
+* A `PeerID` is the textual representation of a peer's `DeviceKey`
+
+In Radicle, keys are securely exchanged with an elliptic curve Diffie-Hellman
```

Comment:
This is no longer accurate -- we are using TLS with a self-signed (by the `DeviceKey`) certificate (until someone comes up with an implementation of noise as the QUIC crypto).